receiver/zipkinreceiver/trace_receiver.go

```diff
@@ -399,35 +398,53 @@ func zipkinSpanToTraceSpan(zs *zipkinmodel.SpanModel) (*tracepb.Span, *commonpb.
 		TimeEvents:   zipkinAnnotationsToProtoTimeEvents(zs.Annotations),
 	}
 
+	node := nodeFromZipkinEndpoints(zs, pbs)
+
 	return pbs, node, nil
 }
 
-func nodeFromZipkinEndpoints(zs *zipkinmodel.SpanModel) *commonpb.Node {
+func nodeFromZipkinEndpoints(zs *zipkinmodel.SpanModel, pbs *tracepb.Span) *commonpb.Node {
 	if zs.LocalEndpoint == nil && zs.RemoteEndpoint == nil {
 		return nil
 	}
 
 	node := new(commonpb.Node)
+	var endpointMap map[string]string
 
 	// Retrieve and make use of the local endpoint
 	if lep := zs.LocalEndpoint; lep != nil {
 		node.ServiceInfo = &commonpb.ServiceInfo{
 			Name: lep.ServiceName,
 		}
-		node.Attributes = zipkinEndpointIntoAttributes(lep, node.Attributes, isLocalEndpoint)
+		endpointMap = zipkinEndpointIntoAttributes(lep, endpointMap, isLocalEndpoint)
 	}
 
 	// Retrieve and make use of the remote endpoint
 	if rep := zs.RemoteEndpoint; rep != nil {
-		// For remoteEndpoint, our goal is to prefix its fields with "zipkin.remoteEndpoint."
-		// For example becoming:
-		// {
-		//      "zipkin.remoteEndpoint.ipv4": "192.168.99.101",
-		//      "zipkin.remoteEndpoint.port": "9000"
-		//      "zipkin.remoteEndpoint.serviceName": "backend",
-		// }
-		node.Attributes = zipkinEndpointIntoAttributes(rep, node.Attributes, isRemoteEndpoint)
+		endpointMap = zipkinEndpointIntoAttributes(rep, endpointMap, isRemoteEndpoint)
 	}
+
+	if endpointMap != nil {
+
+		if pbs.Attributes == nil {
+			pbs.Attributes = &tracepb.Span_Attributes{}
+		}
+		if pbs.Attributes.AttributeMap == nil {
+			pbs.Attributes.AttributeMap = make(map[string]*tracepb.AttributeValue, 6)
```

Comment:
Why 6?

Reply:
Good catch, actually it should be len of the map about to be written in the attribute map. That said I needed to explain the usage of `6` in other location too.